arm/proofs/decode.ml

```diff
@@ -986,7 +1011,11 @@ let rec decode_all = function
   let th1 = READ_WORD_CONV (mk_comb (`read_int32`, tm)) in
   let a,next = dest_pair (rand (rhs (concl th1))) in
   let th = DECODE_CONV (mk_comb (`decode`, a)) in
-  rand (rhs (concl th)) :: decode_all next;;
+  let h = try rand (rhs (concl th))
+    with Failure msg ->
+      let msg' = "Term `" ^ (string_of_term (concl th)) ^ "`: " ^ msg in
+      failwith msg' in
```

Comment:
I slightly updated this code to print the problematic term (`concl th`) by unwrapping `rand (rhs (concl th))` with try-with and printing the term. This helped me quickly find out which instruction was not formalized yet from elf.